### ISSUE TITLE
Should we use a more complex conv nets?

### ISSUE THREAD

**Issue by hlthu** [hlthu/Self-Contained-NN#1]
@GaofengCheng,

Should we use a more complex conv nets, we just use 4 conv layers as feature extractor, and the accuracy is about 72%.

And there are some baselines using VGGNet, ResNet and etc. in https://github.com/hlthu/pytorch-cifar, and the accuracy is more than 92%.

**Comment by GaofengCheng** [comment]
Yes, 73% is far behind the state of art result.

**Comment by hlthu** [comment]
Ok, will use VGG11 firstly.

**Comment by hlthu** [comment]
I have just tried the VGG11 with 200 epochs, the logs and accuracy is as below:

```
Epoch: 10,  loss: 451.301
Epoch: 20,  loss: 451.291
Epoch: 30,  loss: 451.283
Epoch: 40,  loss: 451.272
Epoch: 50,  loss: 451.256
Epoch: 60,  loss: 451.234
Epoch: 70,  loss: 451.195
Epoch: 80,  loss: 451.120
Epoch: 90,  loss: 450.927
Epoch: 100,  loss: 449.844
Epoch: 110,  loss: 394.375
Epoch: 120,  loss: 337.754
Epoch: 130,  loss: 311.056
Epoch: 140,  loss: 294.796
Epoch: 150,  loss: 278.781
Epoch: 160,  loss: 266.682
Epoch: 170,  loss: 254.012
Epoch: 180,  loss: 241.881
Epoch: 190,  loss: 228.402
Epoch: 200,  loss: 215.293
Accuracy of the network on the 10000 test images: 58.00 %
Accuracy of plane : 56.25 %
Accuracy of   car : 92.31 %
Accuracy of  bird : 38.46 %
Accuracy of   cat : 18.18 %
Accuracy of  deer : 15.38 %
Accuracy of   dog : 66.67 %
Accuracy of  frog : 50.00 %
Accuracy of horse : 66.67 %
Accuracy of  ship : 76.19 %
Accuracy of truck : 64.71 %
```

Since the model goes deeper, it becomes slower to converge. I have check the codes in  https://github.com/hlthu/pytorch-cifar, where batchnorm layers are added after every conv layers,also trained for 200 epochs, should I also add the batchnorm layers?

**Comment by hlthu** [comment]
With the VGG11 and also BatchNorm and L2, trained for 100 epochs, the results are:
```
Epoch: 10,  loss: 13.837
Epoch: 20,  loss: 0.198
Epoch: 30,  loss: 0.122
Epoch: 40,  loss: 0.116
Epoch: 50,  loss: 0.111
Epoch: 60,  loss: 0.120
Epoch: 70,  loss: 0.103
Epoch: 80,  loss: 0.103
Epoch: 90,  loss: 0.103
Epoch: 100,  loss: 0.111
Finished Training
Accuracy of the network on the 10000 test images: 82.00 %
Accuracy of plane : 81.25 %
Accuracy of   car : 100.00 %
Accuracy of  bird : 92.31 %
Accuracy of   cat : 40.91 %
Accuracy of  deer : 84.62 %
Accuracy of   dog : 66.67 %
Accuracy of  frog : 66.67 %
Accuracy of horse : 91.67 %
Accuracy of  ship : 95.24 %
Accuracy of truck : 88.24 %
```

**Comment by hlthu** [comment]
VGG16 for 100 epochs:
```
Epoch: 10,  loss: 18.692
Epoch: 20,  loss: 0.270
Epoch: 30,  loss: 0.098
Epoch: 40,  loss: 0.075
Epoch: 50,  loss: 0.080
Epoch: 60,  loss: 0.074
Epoch: 70,  loss: 0.076
Epoch: 80,  loss: 0.076
Epoch: 90,  loss: 0.079
Epoch: 100,  loss: 0.079
Accuracy of the network on the 10000 test images: 85.00 %
Accuracy of plane : 93.75 %
Accuracy of   car : 92.31 %
Accuracy of  bird : 100.00 %
Accuracy of   cat : 59.09 %
Accuracy of  deer : 84.62 %
Accuracy of   dog : 66.67 %
Accuracy of  frog : 77.78 %
Accuracy of horse : 91.67 %
Accuracy of  ship : 90.48 %
Accuracy of truck : 88.24 %
```
